src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/PopupRenderer.ios.cs

```diff
@@ -0,0 +1,288 @@
+		{
+			var view = Element.Content;
+			var contentPage = new ContentPage { Content = view, Padding = new Thickness(25) };
+
+			Control = Platform.CreateRenderer(contentPage);
+			Platform.SetRenderer(contentPage, Control);
+			contentPage.Parent = Application.Current.MainPage;
+		}
+
+		void SetViewController()
+		{
+			var currentPageRenderer = Platform.GetRenderer(Application.Current.MainPage);
+			ViewController = currentPageRenderer.ViewController;
+		}
+
+		void SetEvents()
+		{
+			Element.Dismissed += OnDismissed;
+		}
+
+		void SetSize()
+		{
+			if (!Element.Size.IsZero)
+			{
+				PreferredContentSize = new CGSize(Element.Size.Width, Element.Size.Height);
+			}
+		}
+
+		void SetLayout()
+		{
+			((UIPopoverPresentationController)PresentationController).SourceRect = new CGRect(0, 0, PreferredContentSize.Width, PreferredContentSize.Height);
+
+			if (Element.Anchor == null)
+			{
+				nfloat originX = 0;
+				nfloat originY = 0;
+
+				switch (Element.VerticalOptions.Alignment)
+				{
+					case LayoutAlignment.End:
+						originY = UIScreen.MainScreen.Bounds.Height - PreferredContentSize.Height;
+						break;
+					case LayoutAlignment.Center:
+						originY = (UIScreen.MainScreen.Bounds.Height / 2) - (PreferredContentSize.Height / 2);
+						break;
+				}
+
+				switch (Element.HorizontalOptions.Alignment)
+				{
+					case LayoutAlignment.End:
+						originX = UIScreen.MainScreen.Bounds.Width;
+						break;
+					case LayoutAlignment.Center:
+						originX = UIScreen.MainScreen.Bounds.Width / 2;
+						break;
+				}
+
+				PopoverPresentationController.SourceRect = new CGRect(originX, originY, 0, 0);
+			}
+			else
+			{
+				var view = Platform.GetRenderer(Element.Anchor).NativeView;
+				PopoverPresentationController.SourceView = view;
+				PopoverPresentationController.SourceRect = view.Bounds;
+			}
+		}
+
+		void SetBackgroundColor()
+		{
+			Control.NativeView.BackgroundColor = Element.Color.ToUIColor();
+		}
+
+		void SetView()
+		{
+			base.View.AddSubview(Control.ViewController.View);
+			base.View.Bounds = new CGRect(0, 0, PreferredContentSize.Width, PreferredContentSize.Height);
+			AddChildViewController(Control.ViewController);
+		}
+
+		void SetPresentationController()
+		{
+			((UIPopoverPresentationController)PresentationController).SourceView = ViewController.View;
+
+			// Setting PermittedArrowDirector to 0 breaks the Popover layout. It would be nice if there is no anchor to remove the arrow.
+			((UIPopoverPresentationController)PresentationController).PermittedArrowDirections = UIPopoverArrowDirection.Up;
+			((UIPopoverPresentationController)PresentationController).Delegate = new PopoverDelegate(this);
+		}
+
+		void AddToCurrentPageViewController()
+		{
+			ViewController.PresentViewController(this, true, () => Element.OnOpened());
+		}
+
+		async void OnDismissed(object sender, PopupDismissedEventArgs e)
+		{
+			if (!FormsToolkit.IsiOS9OrNewer)
+				await ViewController.DismissViewControllerAsync(true);
```

Comment:
This makes sense, I think updating it to use the `ConfigureAwait(false)` makes the most sense there may be compiler warnings otherwise. Setting it to false is a good idea since we don't care about what thread it returns on.

Reply:
I decided to remove the async/await api as suggested. Look for it in the next code push